### PR TITLE
Feature/update check icon pr py

### PR DESCRIPTION
**Pull request by Snailedlt**
## Double check these details before you open a PR

- [x] PR does not match another non-stale PR currently opened

## Features

1. Check if name attribute exists and is of the correct type (string)
2. Check if altnames attribute exists, is of correct type (array of strings) and does not contain name
3. Add extra checks to aliases to ensure they contain `base` and `alias, and that they both match the regexp: (original|plain|line)(-wordmark)?

**This PR closes NONE**

### CODE REVIEW

**Review by Thomas-Boi — APPROVED** [review]
Hey @Snailedlt,

The PR looks great 👍 . The checks make sense to me and it seems like we now have a `altname` field in the object. This should be a great feature for the checking script. I'll approve this PR. However, I'm not a maintainer anymore so please double check with the others just in case.

Good luck with future PRs. I think you'll be a great member for the Devicon maintainer team 😄

**Comment by Snailedlt** [comment]
@Panquesito7 please check this out when you have time :)
Should be merged ASAP since it will make reviews easier and quicker

**Review by lunatic-fox — CHANGES_REQUESTED** [review]
I've reviewed it and forgot to finish. 😑

File: .github/scripts/check_icon_pr.py [hunk 1]
```diff
@@ -105,10 +123,35 @@ def check_devicon_object(icon: dict):
...
     except KeyError:
         err_msgs.append("- missing key: 'aliases'.")
-    
+
+    try:
+        for alias_objects in icon["aliases"]:
+            if type(alias_objects) != dict:
+                raise TypeError()
+    except TypeError:
+        err_msgs.append("- 'aliases' must be an array of objects, not: " + str(icon["aliases"]))
```
Comment (Panquesito7) [inline .github/scripts/check_icon_pr.py:135]:
```suggestion
        err_msgs.append("- \"aliases\" must be an array of objects, not: " + str(icon["aliases"]))
```

File: .github/scripts/check_icon_pr.py [hunk 2]
```diff
@@ -105,10 +123,35 @@ def check_devicon_object(icon: dict):
...
+
+    try:
+        for alias_objects in icon["aliases"]:
+            if type(alias_objects["base"]) != str:
+                err_msgs.append("- must contain at least 1 base in aliases.")
+            if not util.is_svg_version_valid(alias_objects['base']):
+                err_msgs.append(f"- Invalid base name in aliases['base']: '{alias_objects['base']}'. Must match regexp: (original|plain|line)(-wordmark)?")
+    except KeyError:
+        err_msgs.append("- missing key: 'base' in 'aliases'.")
```
Comment (Panquesito7) [inline .github/scripts/check_icon_pr.py:144]:
```suggestion
        err_msgs.append("- missing key: \"base\" in \"aliases\".")
```

File: .github/scripts/check_icon_pr.py [hunk 3]
```diff
@@ -105,10 +123,35 @@ def check_devicon_object(icon: dict):
...
+                raise TypeError()
+    except TypeError:
+        err_msgs.append("- 'aliases' must be an array of objects, not: " + str(icon["aliases"]))
+
+    try:
+        for alias_objects in icon["aliases"]:
+            if type(alias_objects["base"]) != str:
+                err_msgs.append("- must contain at least 1 base in aliases.")
+            if not util.is_svg_version_valid(alias_objects['base']):
+                err_msgs.append(f"- Invalid base name in aliases['base']: '{alias_objects['base']}'. Must match regexp: (original|plain|line)(-wordmark)?")
```
Comment (Snailedlt) [inline .github/scripts/check_icon_pr.py:142]:
```suggestion
                err_msgs.append(f"- Invalid base name in aliases[\"base\"]: \"{alias_objects['base']}\". Must match regexp: (original|plain|line)(-wordmark)?")
```

File: .github/scripts/check_icon_pr.py [hunk 4]
```diff
@@ -105,10 +123,35 @@ def check_devicon_object(icon: dict):
...
+    except KeyError:
+        err_msgs.append("- missing key: 'base' in 'aliases'.")
+
+    try:
+        for alias_objects in icon["aliases"]:
+            if type(alias_objects["alias"]) != str:
+                err_msgs.append("- must contain at least 1 alias in aliases.")
+            if not util.is_svg_version_valid(alias_objects['alias']):
+                err_msgs.append(f"- Invalid alias name in aliases['alias']: '{alias_objects['alias']}'. Must match regexp: (original|plain|line)(-wordmark)?")
```
Comment (Snailedlt) [inline .github/scripts/check_icon_pr.py:151]:
```suggestion
                err_msgs.append(f"- Invalid alias name in aliases['alias']: \"{alias_objects['alias']}\". Must match regexp: (original|plain|line)(-wordmark)?")
```

File: .github/scripts/check_icon_pr.py [hunk 5]
```diff
@@ -105,10 +123,35 @@ def check_devicon_object(icon: dict):
...
+    except KeyError:
+        err_msgs.append("- missing key: 'alias' in 'aliases'.")
+
     if len(err_msgs) > 0:
         message = "Error found in 'devicon.json' for '{}' entry: \n{}".format(icon["name"], "\n".join(err_msgs))
```
Comment (Snailedlt) [inline .github/scripts/check_icon_pr.py:156]:
```suggestion
        message = "Error found in \"devicon.json\" for \"{}\" entry: \n{}".format(icon["name"], "\n".join(err_msgs))
```

File: .github/scripts/check_icon_pr.py [hunk 6]
```diff
@@ -105,12 +123,37 @@ def check_devicon_object(icon: dict):
 
     try:
         if type(icon["aliases"]) != list:
-            err_msgs.append("- 'aliases' must be an array.")
+            err_msgs.append("- \"aliases\" must be an array of objects.")
     except KeyError:
         err_msgs.append("- missing key: 'aliases'.")
-    
+
+    try:
```
Comment (Panquesito7) [inline .github/scripts/check_icon_pr.py:130]: Shouldn't we check if the `alias` is the same as an existent icon version? If so, an error must be shown, I guess.

Comment (Snailedlt) [reply .github/scripts/check_icon_pr.py:130]: Yes, but I think @lunatic-fox fixes this in another PR. I can't remember which one though

**Review by Panquesito7 — APPROVED** [review]
Great work! Thanks. 🚀

**Comment by Snailedlt** [comment]
@lunatic-fox if this looks okey to you I can go ahead and merge it to develop :)

**Review by lunatic-fox — APPROVED** [review]

**Pull request merged**